scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -889,6 +894,48 @@ public boolean isConnected() {
         return frontendState.isConnected();
     }
 
+    @Override
+    public String getCurrentPolicy() throws NotConnectedException, PermissionException {
+        return policyFullName;
+    }
+
+    @Override
+    public Map<JobId, JobDescriptor> getJobsToSchedule() throws NotConnectedException, PermissionException {
+        Map<JobId, JobDescriptor> jobMap = schedulingService.lockJobsToSchedule();
+        schedulingService.unlockJobsToSchedule(jobMap.values());
+        return jobMap;
+    }
+
+    @Override
+    public List<TaskDescriptor> getTasksToSchedule() throws NotConnectedException, PermissionException {
+        Policy policy = null;
+        List<TaskDescriptor> eligibleTasks = new ArrayList<>();
+        Map<JobId, JobDescriptor> jobMap = schedulingService.lockJobsToSchedule();
```

Comment:
You should move this instruction in the try bloc. If you leave it there, if there is an exception in lockJobsToSchedule(), the lock will not be released in the finally bloc.